mipengine/node/monetdb_interface/merge_tables.py

```diff
@@ -0,0 +1,47 @@
+from typing import List
+
+from mipengine.node.monetdb_interface import common
+from mipengine.node.monetdb_interface import tables
+from mipengine.node.monetdb_interface.common import connection
+from mipengine.node.monetdb_interface.common import convert_schema_to_sql_query_format
+from mipengine.node.monetdb_interface.common import cursor
+from mipengine.node.tasks.data_classes import TableInfo
+from mipengine.utils.custom_exception import IncompatibleSchemasMergeException
+from mipengine.utils.custom_exception import TableCannotBeFound
+
+
+def get_merge_tables_names(context_id: str) -> List[str]:
+    return common.get_tables_names("merge", context_id)
+
+
+def create_merge_table(table_info: TableInfo):
+    columns_schema = convert_schema_to_sql_query_format(table_info.schema)
+    cursor.execute(f"CREATE MERGE TABLE {table_info.name} ( {columns_schema} )")
+
+
+def get_non_existing_tables(table_names: List[str]) -> List[str]:
+    names_clause = str(table_names)[1:-1]
+    cursor.execute(f"SELECT name FROM tables WHERE name IN({names_clause})")
+    existing_table_names = [table[0] for table in cursor]
+    return [table_name for table_name in table_names + existing_table_names if table_name not in table_names or table_name not in existing_table_names]
+
+
+def add_to_merge_table(merge_table_name: str, partition_tables_names: List[str]):
+    try:
+        non_existing_tables = get_non_existing_tables(partition_tables_names)
+        table_infos = [TableInfo(name, tables.get_table_schema(name)) for name in partition_tables_names]
+        for name in partition_tables_names:
+            cursor.execute(f"ALTER TABLE {merge_table_name} ADD TABLE {name.lower()}")
+
+    except Exception as exc:
+        code = str(exc)[0:5]
+        if code == '3F000':
+            connection.rollback()
+            raise IncompatibleSchemasMergeException(table_infos)
+        elif code == '42S02':
+            connection.rollback()
+            raise TableCannotBeFound(non_existing_tables)
+        else:
+            connection.rollback()
+            raise exc
```

Comment:
1. connection.rollback() should not be in the finally. It shouldn't be executed if an exception doesn't occur.
2. If an exception occured with different code than in the if statements, it will go in the else and reraise the exception, after rolling back. This should not be changed.
3. If any of the functions failed, again it will go in the else statement and rollback before reraising the exception. 
4. I think that it is visible from the exception that is being raised, what the error code means. We could maybe add a comment. What do you think?

Reply:
Oook, I misunderstood, sorry